src/rpc/rawtransaction.cpp

```diff
@@ -123,6 +123,39 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
     }
 }
 
+UniValue getconfirmations(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw runtime_error(
+                "getconfirmations \"txid\"\n"
+                        "\nReturns the number of transaction confirmations.\n"
+                        "\nResult:\n"
+                        "n    (numeric) The number of confirmations\n"
+                        "\nExamples:\n"
+                + HelpExampleCli("getconfirmations", "\"txid\"")
+                + HelpExampleRpc("getconfirmations", "\"txid\"")
+        );
+
+    LOCK(cs_main);
+
+    uint256 hash;
+    hash.SetHex(request.params[0].get_str());
+
+    CTransactionRef tx;
+    uint256 hashBlock;
+    if (!GetTransaction(hash, tx, Params().GetConsensus(), hashBlock, true))
```

Comment:
This requires `txindex=1`, otherwise only works on unconfirmed transactions.

Seems like this ought to be a wallet RPC?